Expose HTMLTextAreaElement.lastChangeWasUserEdit to JS when _WKContentWorldConfiguration.allowAutofill is YES

arenevier opened the pull request:
#### dfa89812d91b8d4140aba543f24429d8c8673c8b
<pre>
Expose HTMLTextAreaElement.lastChangeWasUserEdit to JS when _WKContentWorldConfiguration.allowAutofill is YES
<a href="https://bugs.webkit.org/show_bug.cgi?id=300911">https://bugs.webkit.org/show_bug.cgi?id=300911</a>
<a href="https://rdar.apple.com/145296109">rdar://145296109</a>

Reviewed by Ryosuke Niwa.

API tests for HTMLTextAreaElement.lastChangeWasUserEdit

* Tools/TestWebKitAPI/Tests/WebKitCocoa/UserContentController.mm:
(TEST(WKUserContentController, LastChangeWasUserEdit)):
(TEST(WKUserContentController, LastChangeWasUserEditNonAutofillWorld)):

Canonical link: <a href="https://commits.webkit.org/301810@main">https://commits.webkit.org/301810@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e7bbe6174ca54cecd2e7d71ca800d9c7a80d811d

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/127095 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/46731 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/37844 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/134097 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/78657 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/0cab5b1a-80a9-4702-976f-5bf8e7e7add8) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/128966 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/47347 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/55257 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/96715 "") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/64759 "") | ⏳ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/130043 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/37901 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/113810 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/77225 "Found 1 new API test failure: WPE/TestWebKitWebContext:/webkit/WebKitWebContext/no-web-process-leak (failure)") | | ⏳ 🛠 vision-apple 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/36771 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/31938 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/77489 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/107768 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/32293 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/136623 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/53750 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/41400 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/105235 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/54254 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/110167 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/104924 "Passed tests") | | 
| [  ~~🛠 🧪 merge~~](https://ews-build.webkit.org/#/builders/19/builds/26761 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/50451 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/28877 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/51297 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/19885 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/53682 "Built successfully") | | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/52920 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/56253 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/54678 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

rniwa commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/UserContentController.mm:
```diff
@@ -1437,6 +1437,75 @@ - (void)_webView:(WKWebView *)webView didClickAutoFillButtonWithUserInfo:(id<NSS
+    }];
+    TestWebKitAPI::Util::run(&isDoneEvaluatingScript);
+    isDoneEvaluatingScript = false;
+    EXPECT_WK_STREQ(@"undefined", resultValue.get());
```
Please also assert that `typeof` returns undefined in the main world.

arenevier committed:
Expose HTMLTextAreaElement.lastChangeWasUserEdit to JS when _WKContentWorldConfiguration.allowAutofill is YES
https://bugs.webkit.org/show_bug.cgi?id=300911
rdar://145296109

Reviewed by Ryosuke Niwa.

API tests for HTMLTextAreaElement.lastChangeWasUserEdit

* Tools/TestWebKitAPI/Tests/WebKitCocoa/UserContentController.mm:
(TEST(WKUserContentController, LastChangeWasUserEdit)):
(TEST(WKUserContentController, LastChangeWasUserEditNonAutofillWorld)):

Canonical link: https://commits.webkit.org/301810@main
